### ISSUE TITLE
No audio-video synchronization using dash-live

### ISSUE THREAD

**Issue by Comandoraf** [gpac/gpac#540]
Hi,
I have a problem, I started MP4Box as dash live server:
`MP4Box -dash-live 2000 -profile live -dynamic -segment-name seg_%s -mpd-refresh 4 -rap -out Glass_Half.mpd ../../Glass_Half_1080.mp4#video ../../Glass_Half_1080.mp4#audio ../../Glass_Half_720.mp4#video ../../Glass_Half_540.mp4#video -time-shift 600 -min-buffer 4000`
I used ffmpeg earlier on these files:

```
~/bin/ffmpeg -y -i CONTENT/Tears_Of_Steel/Tears_Of_Steel.mp4 -c:a libfdk_aac -ac 2 -ab 128k -c:v libx264 -x264opts 'keyint=24:min-keyint=24:no-scenecut' -b:v 2400k -maxrate 4800k -bufsize 9600k -vf "scale=-1:1080" Tears_Of_Steel_1080.mp4
~/bin/ffmpeg -y -i CONTENT/Tears_Of_Steel/Tears_Of_Steel.mp4 -c:a libfdk_aac -ac 2 -ab 128k -c:v libx264 -x264opts 'keyint=24:min-keyint=24:no-scenecut' -b:v 1200k -maxrate 2400k -bufsize 4800k -vf "scale=-1:720" Tears_Of_Steel_720.mp4
~/bin/ffmpeg -y -i CONTENT/Tears_Of_Steel/Tears_Of_Steel.mp4 -c:a libfdk_aac -ac 2 -ab 128k -c:v libx264 -x264opts 'keyint=24:min-keyint=24:no-scenecut' -b:v 600k -maxrate 1200k -bufsize 2400k -vf "scale=-1:540" Tears_Of_Steel_540.mp4
```

After few iterations of server MP4Client start lagging, and after reset it (no server) audio is playing back slower then video and with time difference between audio and video is going larger.

**Comment by rbouqueau** [comment]
I've just tried with our counter content to no avail.

Coud your share your source files? We have an anonymous dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

**Comment by Comandoraf** [comment]
https://drive.google.com/folderview?id=0B02_Gt8HMtdtQUprcTEtdHU5YlE&usp=sharing

**Comment by rbouqueau** [comment]
Thank you

2 remarks:
1) audio and video do not have the closest possible durations
2) there are edit-lists on both audio and video and that's a know source of wrong durations with MP4box -dash-live

**Comment by Comandoraf** [comment]
So is it a bug, or should I do something different way?

**Comment by hanyufeng520** [comment]
@Comandoraf  Hello, Can MP4box be used to live streaming ?For example,I have one frame 40 per ms,how can I streaming these frames for a long time ?

**Comment by Comandoraf** [comment]
I don't understand your question. You can use MP4Box as a server of DASH live stream in way I wrote in first post. After end of video server starts generate it from begining.

**Comment by hanyufeng520** [comment]
@Comandoraf Thanks for your reply.My question is:
I have one mp4 file every ten seconds. For example  0s- 10s , I have 1.mp4, 10s-20s I have 2.mp4,
20s-30s I have 3.mp4 ....

So when I have 1.mp4 ,I need to dash it.After I start a client to watch the streaming ....
when I have 2.mp4 ,I also need to dash it.  
when I have 3.mp4 ,I also need to dash it.
when I have 4.mp4 ,I also need to dash it.
....................................................

I only want to start the client once ,and then I just watch the streaming  for long time(for example 10 hours, so I need to watch about 10*360 = 3600 mp4 files).

In your example ,you have all the mp4 files need .But in my case ,I only have the first MP4.
and after 2.MP4 ,  3.mp4, 4.mp4....

Do you have any idea about this?Thanks.

**Comment by Comandoraf** [comment]
So you have any chunk generator and only thing you need is mpd?

**Comment by hanyufeng520** [comment]
@Comandoraf  Yes.When I have one more ts file ,I use MP4box to chunk it.So the target is I start the client only once ,and I can see the streaming content of all the chunked content.

**Comment by Comandoraf** [comment]
You are using MP4Box every 10s? Well, in fact MP4Box I think can't be used in that way, because it generates mpd and chunks also. But you only need mpd which you can write even by yourself(what I understood) :) You need that line of mpd: <SegmentTemplate timescale="XXX" media="$Number$.mp4" startNumber="1" duration="XXX"/>. But I am not an expert in MP4Box so ask maybe @rbouqueau.
And I ask again about my problem, can I do something to play it correctly or I have to wait for fix?

**Comment by rbouqueau** [comment]
@hanyufeng520 please describe your issue more clearly at #542

**Comment by ulrichghislain** [comment]
Hello i need a help please , i'am a beginner on dashcast 
i have a stream mpeg2ts and i want to have a stream dashcast 
this is command that i'm typing
"DashCast -av udp://@239.x.x.x:1234?fifo_size=3000000 -live -seg-dur 8000 -mpd-refresh 8 -time-shift -1 -min-buffer 10 -npts -conf config_dashcast.conf -ast-offset 16000" then i just have a folder output empty (no chunk stream inside)

i get this error in command ---->"[mpegts @ 0x24561e0] Could not find codec parameters for stream 6 (Unknown: none ([5][0][0][0] / 0x0005)): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options"

----->"Assertion desc failed at libswscale/swscale_internal.h:674
thanks in advance

**Comment by rbouqueau** [comment]
@ulrichghislain Stop cross-posting, especially on un-related issues. This is not respectful of the other users.

**Comment by ulrichghislain** [comment]
soory ,i 'm disperate

**Comment by jeanlf** [comment]
this should now be fixed on git, see #885
